py/modmath.c

```diff
@@ -171,6 +171,45 @@ MATH_FUN_1(lgamma, lgamma)
 #endif
 //TODO: fsum
 
+static inline mp_float_t get_parsed_float(const mp_obj_t parsed_val, mp_float_t def) {
+    return parsed_val == MP_OBJ_NULL ? def : mp_obj_get_float(parsed_val);
+}
+
+STATIC mp_obj_t mp_math_isclose(size_t n_args, const mp_obj_t *args, mp_map_t *kwargs) {
+    enum { ARG_a, ARG_b, ARG_rel_tol, ARG_abs_tol };
+    static const mp_arg_t allowed_args[] = {
+        {MP_QSTR_, MP_ARG_REQUIRED | MP_ARG_OBJ},
+        {MP_QSTR_, MP_ARG_REQUIRED | MP_ARG_OBJ},
+        {MP_QSTR_rel_tol, MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_obj = MP_OBJ_NULL}},
+        {MP_QSTR_abs_tol, MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_obj = MP_OBJ_NULL}},
+    };
+    mp_arg_val_t parsed_args[MP_ARRAY_SIZE(allowed_args)];
+    mp_arg_parse_all(n_args, args, kwargs, MP_ARRAY_SIZE(allowed_args), allowed_args, parsed_args);
+    const mp_float_t a = mp_obj_get_float(parsed_args[ARG_a].u_obj);
+    const mp_float_t b = mp_obj_get_float(parsed_args[ARG_b].u_obj);
+    const mp_float_t rel_tol = get_parsed_float(parsed_args[ARG_rel_tol].u_obj, (mp_float_t)1e-9);
+    const mp_float_t abs_tol = get_parsed_float(parsed_args[ARG_abs_tol].u_obj, (mp_float_t)0.0);
+
+    if (rel_tol < (mp_float_t)0.0 || abs_tol < (mp_float_t)0.0) {
+        math_error();
+    }
+    if (a == b) {
+        return mp_const_true;
+    }
+    if (isinf(a) || isinf(b)) {
+        return mp_const_false;
+    }
+
+    const mp_float_t difference = MICROPY_FLOAT_C_FUN(fabs)(a - b);
```

Comment:
If this goes before the `isinf` tests then it should be possible to test that either a or b are inf by a single call like `isinf(difference)`, to save code size and make it more efficient for the more common case where inf is not passed in.